lib/src/test/java/io/ably/lib/test/realtime/RealtimePresenceTest.java

```diff
@@ -1649,6 +1649,125 @@ public void onPresenceMessage(PresenceMessage message) {
         }
     }
 
+    private final class CountingCompletionListener implements CompletionListener
```

Comment:
you could put this in with the helpers

Reply:
Ah I wasn't aware those existed - there's already a listener in there that is essentially the same, but without timeouts. I have adapted it to allow for timeouts and used that!